paasta_tools/setup_marathon_job.py

```diff
@@ -172,6 +172,8 @@ def drain_tasks_and_find_tasks_to_kill(
             all_draining_tasks.add((task, client))
 
     for task, client in all_draining_tasks:
```

Comment:
Whoa, isn't this loop really wasteful, especially in large task lists?

We are iterating over it twice on line 185?

Can you combine them so we only have to drain/check once?

Past that, is UNREACH the only state where we can do this? How instead we only do the draining if it is RUNNING?